guilford_dps_offenders.yaml

```diff
@@ -0,0 +1,330 @@
+        type: "ol.weight"
+        column: "OFFENDER WEIGHT (IN LBS)"
+      nc.complexion:
+        type: "nc.complexion"
+        # column: "OFFENDER SKIN COMPLEXION CODE"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "OFFENDER SKIN COMPLEXION CODE"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "OFFENDER SKIN COMPLEXION CODE"
+          - !<transforms.ReplaceTransform>
+            target: ["RUDDY", "DARK", "FRECKLED", "LIGHT", "UNKNOWN", "ALBINO", "MEDIUM"]
+            goal: ["ruddy", "dark", "freckled", "light", "unknown", "albino", "medium"]
+      ol.haircolor:
+        type: "ol.haircolor"
+        # column: "OFFENDER HAIR COLOR CODE"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "OFFENDER HAIR COLOR CODE"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "OFFENDER HAIR COLOR CODE"
+          - !<transforms.ReplaceTransform>
+            target: ["RED", "WHITE", "BLONDE", "BLACK", "SANDY", "OTHER", "UNKNOWN", "BROWN", "BALD", "GRAY"]
+            goal: ["red", "white", "blonde", "black", "sandy", "other", "unknown", "brown", "bald", "gray"]
+      ol.bodydescription:
+        type: "ol.bodydescription"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "OFFENDER BODY BUILD CODE"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "OFFENDER BODY BUILD CODE"
+      ol.eyecolor:
+        type: "ol.eyecolor"
+        # column: "OFFENDER EYE COLOR CODE"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "OFFENDER EYE COLOR CODE"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "OFFENDER EYE COLOR CODE"
+          - !<transforms.ReplaceTransform>
+            target: ["AMBER", "BLACK", "VIOLET", "HAZEL", "OTHER", "MIXED", "BROWN", "GRAY", "GREEN", "BLUE"]
+            goal: ["amber", "black", "violet", "hazel", "other", "mixed", "brown", "gray", "green", "blue"]
+    name: "offenderappearance"
```

Comment:
can you add a condition on this entity set to only read into it if at least one column in it is not null? So it would start with a boolean or condition.

Reply:
done